pkgs/tools/admin/pgadmin/expose-setup.py.patch

```diff
@@ -25,41 +19,42 @@ index 5592d1b04..131eaa1ed 100644
  
  )
 diff --git a/web/setup.py b/web/setup.py
-index 5f4257e86..df970f049 100644
+index 2204ffb..d5fda9f 100644
 --- a/web/setup.py
 +++ b/web/setup.py
-@@ -32,6 +32,10 @@ from pgadmin import create_app
- from pgadmin.utils import clear_database_servers, dump_database_servers,\
-     load_database_servers
- 
-+# Configuration settings
-+import config
-+from pgadmin.model import SCHEMA_VERSION
-+from pgadmin.setup import db_upgrade, create_app_data_directory
- 
- def dump_servers(args):
-     """Dump the server groups and servers.
-@@ -139,12 +143,7 @@ def clear_servers():
+@@ -14,7 +14,6 @@ import argparse
+ import os
+ import sys
+ import builtins
+-import config
+
+ # Grab the SERVER_MODE if it's been set by the runtime
+ if 'SERVER_MODE' in globals():
+@@ -91,6 +90,9 @@ def load_servers(args):
+ def setup_db(app):
+     """Setup the configuration database."""
+
++    # here we need an additional import, since the call to setup_db changed
++    # https://redmine.postgresql.org/projects/pgadmin4/repository/2/revisions/3a69d50458f57d19948b867aec55c55d635a36e5/diff/web/setup.py
++    import config
```

Comment:
It is not enough to import it in line 29 instead?

Reply:
The import in line 29 was moved in pgadmin4 6.12 from the main routine to the global space. Unfortunately, the nixos module pf pgadmin fails with an importError, since it cannot find `config`. I suspect, its because `config` is local to the `setup.py` file and when imported globally, it fails to find it. 
It works when not imported at the top of the file, since the normal call to the script doesn't import any local source files. They changed the call from `pgadmin4.py` to the `setup_db.py`. For reference, this was the call to the script before:
```
# Check if the database exists. If it does not, create it.
if not os.path.isfile(config.SQLITE_PATH):
setup_py = os.path.join(
os.path.dirname(os.path.realpath(u_encode(__file__, fs_encoding))),
'setup.py'
)
exec(open(file_quote(setup_py), 'r').read())
```

Now, it is just called with:
```
# Check if the database exists. If it does not, create it.
--
if not os.path.isfile(config.SQLITE_PATH):
setup.setup_db(app)
```

This is why `import config` needs to be in the `setup_db` routine and needs to be removed from the global import for nixpkgs.